test/component_test.go

```
@@ -0,0 +1,68 @@
+			autoscalingGroupId := atm.Output(component, "autoscaling_group_id")
+			assert.True(t, strings.HasPrefix(autoscalingGroupId, iamInstanceProfile))
+
+			securityGroupId := atm.Output(component, "security_group_id")
+			assert.True(t, strings.HasPrefix(securityGroupId, "sg-"))
+
+			instanceIds := aws.GetInstanceIdsForAsg(t, autoscalingGroupId, awsRegion)
+			assert.Equal(t, 1, len(instanceIds))
+
+			instance := GetEc2Instances(t, instanceIds[0], awsRegion)
+			assert.EqualValues(t, "t2.micro", instance.InstanceType)
+			assert.EqualValues(t, "running", *&instance.State.Name)
```

golangci-lint (1.62.2)

SA4001: *&x will be simplified to x. It will not copy x.

(staticcheck)